### ISSUE TITLE
Identify offending package when ClassMapGenerator cannot scan for classes

### ISSUE THREAD

**Issue by TravisCarden** [composer/composer#8328]
tl;dr: In order to simplify debugging, Composer should identify the offending package when `ClassMapGenerator` cannot scan for classes.

If a package specifies a non-existent path in its "classmap", Composer fails with output like the below that doesn't identify the offending package. I include it in its entirety to illustrate that the culprit cannot necessarily be inferred from it. In this case, it's the third from the last package operation. The only way to identify it is to grep `composer.json` files in the `vendor` directory for the missing path, which is tedious at best. It would greatly simplify debugging if the error message indicated the source of the bad path, e.g., `Could not scan for classes inside "vendor/drupal/acm_sku/tests/src/" which does not appear to be a file nor a folder`. (Merely adding the full path relative to the project root eliminates all ambiguity.)

```
$ composer init -n
$ composer config repositories.drupal composer https://packages.drupal.org/8
$ composer config minimum-stability dev
$ composer require drupal/acm_sku drupal/example
    1/1:	http://repo.packagist.org/p/provider-latest$5a20935d40ca35b875fe4341d4c18ce349e903dbdf71ef7f419f310c9e902c36.json
    Finished: success: 1, skipped: 0, failure: 0, total: 1
Using version 1.x-dev for drupal/acm_sku
Using version 3.x-dev for drupal/example
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 76 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-mbstring (dev-master b42a2f6): Cloning b42a2f66e8 from cache
  - Installing paragonie/random_compat (v9.99.99.x-dev 0947f25): Cloning 0947f25b88 from cache
  - Installing symfony/polyfill-php70 (dev-master 54b4c42): Cloning 54b4c428a0 from cache
  - Installing symfony/http-foundation (3.4.x-dev 089da40): Cloning 089da40dfa from cache
  - Installing psr/http-message (dev-master efd67d1): Cloning efd67d1dc1 from cache
  - Installing symfony/psr-http-message-bridge (dev-master 81ae86d): Cloning 81ae86d339 from cache
  - Installing brumann/polyfill-unserialize (v1.0.4): Loading from cache
  - Installing typo3/phar-stream-wrapper (v2.x-dev 057622f): Cloning 057622f5a3 from cache
  - Installing psr/container (dev-master 014d250): Cloning 014d250dae from cache
  - Installing symfony/dependency-injection (3.4.x-dev 2709bc2): Cloning 2709bc2978 from cache
  - Installing symfony/polyfill-ctype (dev-master 550ebaa): Cloning 550ebaac28 from cache
  - Installing twig/twig (1.x-dev 144c557): Cloning 144c557bb1 from cache
  - Installing pear/pear_exception (dev-master 8c18719): Cloning 8c18719fda from cache
  - Installing pear/console_getopt (v1.4.2): Loading from cache
  - Installing pear/pear-core-minimal (v1.10.9): Loading from cache
  - Installing pear/archive_tar (dev-master 7e48add): Cloning 7e48add6f8 from cache
  - Installing psr/log (dev-master c4421fc): Cloning c4421fcac1 from cache
  - Installing symfony/debug (4.4.x-dev 01028c1): Cloning 01028c1cfd from cache
  - Installing symfony/event-dispatcher (3.4.x-dev 3e922c4): Cloning 3e922c4c34 from cache
  - Installing symfony/http-kernel (3.4.x-dev c577c0d): Cloning c577c0ddf4 from cache
  - Installing asm89/stack-cors (dev-master f45a62f): Cloning f45a62f9da from cache
  - Installing composer/semver (dev-master 37e4db2): Cloning 37e4db276f from cache
  - Installing zendframework/zend-diactoros (dev-release-1.8 e03f490): Cloning e03f490426 from cache
  - Installing masterminds/html5 (2.x-dev cadcfaa): Cloning cadcfaaa13 from cache
  - Installing doctrine/lexer (dev-master 6c4b4c3): Cloning 6c4b4c3432 from cache
  - Installing egulias/email-validator (dev-master 950d066): Cloning 950d0663dc from cache
  - Installing stack/builder (dev-master 68e73a9): Cloning 68e73a9e29 from cache
  - Installing zendframework/zend-stdlib (dev-develop 519a966): Cloning 519a966ded from cache
  - Installing zendframework/zend-escaper (dev-develop 14166bb): Cloning 14166bbd4e from cache
  - Installing zendframework/zend-feed (dev-develop 07d0ae6): Cloning 07d0ae6c11 from cache
  - Installing easyrdf/easyrdf (0.9.x-dev acd09df): Cloning acd09dfe05 from cache
  - Installing symfony/routing (3.4.x-dev 8b0faa6): Cloning 8b0faa681c from cache
  - Installing symfony-cmf/routing (1.4.x-dev fb1e7f8): Cloning fb1e7f85ff from cache
  - Installing ralouphie/getallheaders (3.0.3): Loading from cache
  - Installing guzzlehttp/psr7 (1.x-dev 2595b33): Cloning 2595b33c1c from cache
  - Installing guzzlehttp/promises (dev-master 17d36ed): Cloning 17d36ed176 from cache
  - Installing guzzlehttp/guzzle (dev-master a7010cc): Cloning a7010cc9b5 from cache
  - Installing doctrine/annotations (1.7.x-dev fa4c4e8): Cloning fa4c4e861e from cache
  - Installing doctrine/reflection (dev-master 4a97a22): Cloning 4a97a22032 from cache
  - Installing doctrine/event-manager (dev-master e05ae70): Cloning e05ae70c63 from cache
  - Installing doctrine/collections (1.6.x-dev c5e0bc1): Cloning c5e0bc17b1 from cache
  - Installing doctrine/cache (dev-master c29471b): Cloning c29471b93a from cache
  - Installing doctrine/persistence (1.2.x-dev 43526ae): Cloning 43526ae633 from cache
  - Installing doctrine/inflector (1.3.x-dev 45d9b13): Cloning 45d9b132b2 from cache
  - Installing doctrine/common (dev-master 8401ee9): Cloning 8401ee9d82 from cache
  - Installing symfony/yaml (3.4.x-dev 768f817): Cloning 768f817446 from cache
  - Installing symfony/polyfill-iconv (dev-master 685968b): Cloning 685968b11e from cache
  - Installing symfony/process (3.4.x-dev d822cb6): Cloning d822cb6540 from cache
  - Installing symfony/translation (3.4.x-dev 49a884e): Cloning 49a884e9ac from cache
  - Installing symfony/validator (3.4.x-dev 0b3815c): Cloning 0b3815cc5c from cache
  - Installing symfony/serializer (3.4.x-dev c1f9d0a): Cloning c1f9d0a197 from cache
  - Installing symfony/console (3.4.x-dev 97b6b55): Cloning 97b6b55889 from cache
  - Installing symfony/class-loader (dev-master 2141b76): Cloning 2141b7653a from cache
  - Installing drupal/core (8.8.x-dev d9c81a7): Cloning d9c81a75a3 from cache
  - Installing defuse/php-encryption (v2.2.1): Loading from cache
  - Installing lcobucci/jwt (3.3.x-dev a11ec5f): Cloning a11ec5f4b4 from cache
  - Installing league/event (dev-master d2cc124): Cloning d2cc124cf9 from cache
  - Installing drupal/consumers (dev-1.x 898abd9): Cloning 898abd9b27 from cache
  - Installing league/oauth2-server (7.4.0): Loading from cache
  - Installing drupal/simple_oauth (dev-3.x f109c8a): Cloning f109c8ab7d from cache
  - Installing drupal/pcb (dev-1.x eee7ec1): Cloning eee7ec1ba1 from cache
  - Installing drupal/key_value_field (dev-1.x c5e3b5b): Cloning c5e3b5b965 from cache
  - Installing drupal/inline_entity_form (dev-1.x 8c8a0e1): Cloning 8c8a0e1f16 from cache
  - Installing drupal/field_group (dev-3.x b245b85): Cloning b245b85f99 from cache
  - Installing commerceguys/addressing (dev-master 2d048bf): Cloning 2d048bfdb3 from cache
  - Installing drupal/address (dev-1.x 9cbec9f): Cloning 9cbec9f242 from cache
  - Installing commerceguys/intl (dev-master 0fc1761): Cloning 0fc176165f from cache
  - Installing sarciszewski/php-future (0.4.2): Loading from cache
  - Installing acquia/http-hmac-php (3.4.0): Loading from cache
  - Installing drupal/acquia_commercemanager (dev-1.x 194eac3): Cloning 194eac35f0 from cache
  - Installing drupal/simple_oauth_extras (dev-3.x)
  - Installing drupal/acm (dev-1.x)
  - Installing drupal/acm_sku_stock (dev-1.x)
  - Installing drupal/acm_sku (dev-1.x)
  - Installing drupal/coder (dev-3.x 2a4dbff): Cloning 2a4dbff940 from cache
  - Installing drupal/example (dev-3.x)
symfony/psr-http-message-bridge suggests installing nyholm/psr7 (For a super lightweight PSR-7/17 implementation)
symfony/dependency-injection suggests installing symfony/config
symfony/dependency-injection suggests installing symfony/finder (For using double-star glob patterns or when GLOB_BRACE portability is required)
symfony/dependency-injection suggests installing symfony/expression-language (For using expressions in service container configuration)
symfony/dependency-injection suggests installing symfony/proxy-manager-bridge (Generate service proxies to lazy load them)
pear/archive_tar suggests installing ext-xz (Lzma2 compression support.)
symfony/http-kernel suggests installing symfony/browser-kit
symfony/http-kernel suggests installing symfony/config
symfony/http-kernel suggests installing symfony/finder
symfony/http-kernel suggests installing symfony/var-dumper
zendframework/zend-feed suggests installing zendframework/zend-http (Zend\Http for PubSubHubbub, and optionally for use with Zend\Feed\Reader)
zendframework/zend-feed suggests installing zendframework/zend-servicemanager (Zend\ServiceManager component, for easily extending ExtensionManager implementations)
zendframework/zend-feed suggests installing zendframework/zend-cache (Zend\Cache component, for optionally caching feeds between requests)
zendframework/zend-feed suggests installing zendframework/zend-db (Zend\Db component, for use with PubSubHubbub)
zendframework/zend-feed suggests installing zendframework/zend-validator (Zend\Validator component, for validating email addresses used in Atom feeds and entries when using the Writer subcomponent)
easyrdf/easyrdf suggests installing ml/json-ld (~1.0)
symfony/routing suggests installing symfony/config (For using the all-in-one router or any loader)
symfony/routing suggests installing symfony/expression-language (For using expression matching)
guzzlehttp/psr7 suggests installing zendframework/zend-httphandlerrunner (Emit PSR-7 responses)
doctrine/cache suggests installing alcaeus/mongo-php-adapter (Required to use legacy MongoDB driver)
symfony/translation suggests installing symfony/config
symfony/validator suggests installing psr/cache-implementation (For using the metadata cache.)
symfony/validator suggests installing symfony/intl
symfony/validator suggests installing symfony/config
symfony/validator suggests installing symfony/expression-language (For using the Expression validator)
symfony/validator suggests installing symfony/property-access (For accessing properties within comparison constraints)
symfony/serializer suggests installing psr/cache-implementation (For using the metadata cache.)
symfony/serializer suggests installing symfony/property-info (To deserialize relations.)
symfony/serializer suggests installing symfony/config (For using the XML mapping loader.)
symfony/serializer suggests installing symfony/property-access (For using the ObjectNormalizer.)
symfony/console suggests installing symfony/lock
symfony/class-loader suggests installing symfony/polyfill-apcu (For using ApcClassLoader on HHVM)
acquia/http-hmac-php suggests installing silex/silex (~1.3.0)
acquia/http-hmac-php suggests installing symfony/security (~3.0.0)
Writing lock file
Generating autoload files

Installation failed, reverting ./composer.json to its original content.


  [RuntimeException]
  Could not scan for classes inside "tests/src/" which does not appear to be a file nor a folder


require [--dev] [--prefer-source] [--prefer-dist] [--no-progress] [--no-suggest] [--no-update] [--no-scripts] [--update-no-dev] [--update-with-dependencies] [--update-with-all-dependencies] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--] [<packages>]...
```

**Comment by thicolares** [comment]
Hey! I would like to open a PR to this one. Can I take it?

@TravisCarden, that was a nice description, by the way.

**Comment by TravisCarden** [comment]
Thanks, @thicolares. :) I don't know if you were asking me, but I don't think there's any reason you shouldn't submit a PR. I think the "Good First Issue" label is more or less meant to invite it.

**Comment by thicolares** [comment]
Sure! Actually, my intention was to avoid more the one person to work on the same thing. :)

**Comment by TravisCarden** [comment]
It's all yours, @thicolares. ;)

**Comment by Seldaek** [comment]
This is really not all that easy to fix without a big refactoring of the autoload dumper, so I'm gonna close as wontfix, because the problem isn't that common. Typically errors happen in your package as dependencies shouldn't be published in such a broken state.

**Comment by thicolares** [comment]
Hi @Seldaek, I was about to say the same thing. I managed to make a possible fix, but it was hard to test the impacts. Good call and thank you.
